persist/sqlite/sql.go

```diff
@@ -0,0 +1,232 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"math/rand"
+	"strings"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3" // import sqlite3 driver
+	"go.uber.org/zap"
+)
+
+const (
+	longQueryDuration = 10 * time.Millisecond
+	longTxnDuration   = 10 * time.Millisecond
+)
+
+type (
+	// A scanner is an interface that wraps the Scan method of sql.Rows and sql.Row
+	scanner interface {
+		Scan(dest ...any) error
+	}
+
+	// A txn is an interface for executing queries within a transaction.
+	txn interface {
```

Comment:
what are these interfaces used for?

Reply:
It hides the commit and rollback functions from the *sql.Tx to enforce transaction handling.

Reply:
ok, `scanner` makes sense to me. `txn` looks like it's only being implemented by `loggedTxn` (?) so might not be strictly necessary.